mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -2861,16 +2866,39 @@ def __update_name(self, name) -> None:
             parent_dir = os.path.dirname(new_file_path)
             os.makedirs(parent_dir, exist_ok=True)
 
-            os.rename(old_file_path, new_file_path)
+            if detach:
+                """"
+                Detaching a block creates a copy of the block file while keeping the existing block
+                file the same. Without detaching a block, the existing block file is simply renamed.
+                """
+                with open(new_file_path, 'w') as f:
+                    f.write(self.content)
```

Comment:
I would suggest you getting the file content before updating the uuid. 
If the file content was not read before, `self.content` will read the file content from the new path with the new uuid.
https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_preparation/models/block/__init__.py#L360-L361
https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_preparation/models/block/__init__.py#L527-L531

Reply:
got it, ty